src/components/CustomSideBar.tsx

```diff
@@ -27,16 +27,18 @@ const CustomSideBar = (props: CustomSideBarProps) => {
         blockScroll
         className={`custom-sidebar relative ${
           props.themeClasses ? props.themeClasses : ""
-        }`}
+        }  ${props.sideBarWidthClass ? props.sideBarWidthClass : "!w-50w"}`}
         showCloseIcon={false}
       >
         <CustomButton
-          className={`custom-sidebar-close-btn ${getCloseBtnLayoutClasses()}`}
+          className={`custom-sidebar-close-btn z-10 ${getCloseBtnLayoutClasses()} ${
+            props.sideBarWidthClass ? props.sideBarWidthClass : "!w-50w"
+          }`}
           onClick={() => props.setVisible(false)}
         >
           {LABELS.CLOSE}
         </CustomButton>
-        <div className="flex flex-col">
+        <div className="flex flex-col z-20">
```

Comment:
why 20 ?

Reply:
Sidebar close button is placed as a fixed element inside .p-sidebar-content which also contains the children of the sidebar that we pass as prop. To keep the alignment of the close button as in the wireframe, provided the same width for this close button (dynamic width class we are passing as prop) and make a right-8 prop. Also to make the proper layout of children, we have z-10 for the close button and have z-20 for the children. so that close button will always be behind of the children of sidebar.  

If we omit the fixed position and use it as a 'relative' positioned element, the close button will always fall behind the overlay of the sidebar. Thats why I designed this as explained above. Please share your thoughts.

Reply:
Try using z-0 and z-1 instead of z-10 and z-20 ?

Reply:
done